alfresco-transformer-base/src/test/java/org/alfresco/transformer/AbstractTransformerControllerTest.java

```diff
@@ -287,4 +299,134 @@ public void testEmptyPojoTransform() throws Exception
         // Assert the reply
         assertEquals(BAD_REQUEST.value(), transformReply.getStatus());
     }
+
+    @Test
+    public void testGetTransformConfigInfo() throws Exception
+    {
+        TransformConfig expectedTransformConfig = objectMapper
+            .readValue(new ClassPathResource("engine_config.json").getFile(),
+                TransformConfig.class);
+
+        ReflectionTestUtils
+            .setField(AbstractTransformerController.class, "ENGINE_CONFIG", "engine_config.json");
+
+        String response = mockMvc.perform(MockMvcRequestBuilders.get("/info"))
+            .andExpect(status().is(OK.value())).andExpect(
+                header().string(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_UTF8_VALUE))
+            .andReturn().getResponse().getContentAsString();
+
+        TransformConfig transformConfig = objectMapper.readValue(response, TransformConfig.class);
+
+        assertEquals(expectedTransformConfig, transformConfig);
+    }
+
+    @Test
+    public void testGetInfoFromConfigWithDuplicates() throws Exception
```

Comment:
Is this a negative test ? If so shouldn't it result in an error ? 

Or otherwise, will the T-Engine show warnings in the log (that indicate duplicates in the config) ?

Reply:
No, this is the way the serialization libraries (_Jackson_, in our case) usually work: they deduplicate entries when deserializing a **JSON array** into a **Java Set<>**. They don't throw any errors.

Reply:
Hmm - we may need a way (in future ?) to provide some level of *semantic* validation to help custom configurations.

Reply:
It might not be worth it. Unless if you see this scenario happening often and causing problems, I don't see it as a priority.
We're definitely not implementing it in this story, as that validation would take quite a bit of work (it doesn't match our Java model, we'd have to parse the `JSONObject`s).

Reply:
Ok, let's see if it becomes an issue. In the meantime, we should document the current behaviour (as demonstrated by the tests).